### PR TITLE
RWA-982-upgrade

### PR DESCRIPTION
**Pull request by Lee365**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-982

### Change description ###
RWA-982 - Upgrade to Springboot 2.6

**Does this PR introduce a breaking change?** (check one with "x")

```
[] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/exceptions/TestFeignClientException.java [hunk 1]
```diff
@@ -10,7 +10,7 @@ public TestFeignClientException(int status, String message) {
     }
 
     public TestFeignClientException(int status, String message, byte[] responseBody) {
-        super(status, message, responseBody);
+        super(status, message, responseBody, null);
```
Comment (raghera) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/exceptions/TestFeignClientException.java:13]: What is the `null` for here?  Should we be populating with something?

**Review by raghera — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/config/SwaggerPublisherTest.java [hunk 2]
```diff
@@ -14,10 +14,12 @@
 
 class SwaggerPublisherTest extends SpringBootIntegrationBaseTest {
 
+    private static final String SWAGGER_DOCS_VERSION = "/v3/api-docs";
```
Comment (raghera) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/config/SwaggerPublisherTest.java:17]: Can we rename this to be consistent with the Case Event Handler please?  Think it is called `OpenAPIPublisher`.

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/config/SwaggerConfiguration.java [hunk 3]
```diff
@@ -1,52 +1,40 @@
...
-import springfox.documentation.spi.DocumentationType;
-import springfox.documentation.spring.web.plugins.Docket;
-import springfox.documentation.swagger2.annotations.EnableSwagger2;
-
-import static java.util.Arrays.asList;
+import org.springframework.web.method.HandlerMethod;
 
 @Configuration
-@EnableSwagger2
 public class SwaggerConfiguration {
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/config/SwaggerConfiguration.java:14]: Can we rename this file to be consistent with the updated Case Event Handler please - `OpenAPIConfiguration`

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/exceptions/TestFeignClientException.java (hunk 1 above)
Comment (satvinder-hullait) [reply src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/exceptions/TestFeignClientException.java]: I think this should be `Collections.emptyMap()`

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/config/SwaggerConfiguration.java (hunk 3 above)
Comment (Lee365) [reply src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/config/SwaggerConfiguration.java]: File renamed to OpenAPIConfiguration.java

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/config/SwaggerPublisherTest.java (hunk 2 above)
Comment (Lee365) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/config/SwaggerPublisherTest.java:17]: File renamed to OpenAPIPublisherTest

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/exceptions/TestFeignClientException.java (hunk 1 above)
Comment (Lee365) [reply src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/exceptions/TestFeignClientException.java]: Passed in responseHeaders

**Review by satvinder-hullait — APPROVED** [review]

**Pull request merged**